pallas-network/src/multiplexer.rs

```diff
@@ -63,112 +65,126 @@ pub struct Segment {
 }
 
 pub enum Bearer {
-    Tcp(TcpStream),
+    Tcp(tcp::TcpStream),
 
     #[cfg(unix)]
-    Unix(UnixStream),
-
-    #[cfg(windows)]
-    NamedPipe(NamedPipeClient),
+    Unix(unix::UnixStream),
+    // #[cfg(windows)]
+    // NamedPipe(NamedPipeClient),
 }
 
-const BUFFER_LEN: usize = 1024 * 10;
-
 impl Bearer {
-    pub async fn connect_tcp(addr: impl ToSocketAddrs) -> Result<Self, tokio::io::Error> {
-        let stream = TcpStream::connect(addr).await?;
-        // add tcp_keepalive
+    fn configure_tcp(stream: &tcp::TcpStream) -> IOResult<()> {
         let sock_ref = socket2::SockRef::from(&stream);
         let mut tcp_keepalive = socket2::TcpKeepalive::new();
         tcp_keepalive = tcp_keepalive.with_time(tokio::time::Duration::from_secs(20));
         tcp_keepalive = tcp_keepalive.with_interval(tokio::time::Duration::from_secs(20));
-        let _ = sock_ref.set_tcp_keepalive(&tcp_keepalive);
-        // add tcp_nodelay
-        let _ = sock_ref.set_nodelay(true);        
+        sock_ref.set_tcp_keepalive(&tcp_keepalive)?;
```

Comment:
Were you going to move the socket2 tcp_keepalive stuff behind a feature flag and the nodelay to set with tokio?  Or was that in a future PR?

Reply:
future PR, this one is already a big one